backend/src/documents/write_document.py

```diff
@@ -2,33 +2,21 @@
 from urllib import parse
 
 from src import context
+from src.database.data.document_item import DocumentItem, document_item_to_dict
 from src.database.database import Database
 
 
 @context.inject
 def write_document(document_id: str, document_url: str, database: Database = None):
-    document_to_store = {
-        "document_id": document_id,
-        "document_url": document_url,
-        "status": "processing",
-    }
-
-    database.write_document(document_to_store)
+    document_item = DocumentItem(document_id, document_url)
+    database.write_document(document_item_to_dict(document_item))
 
 
 @context.inject
 def update_document(document_url: str, document_type: str | None, extracted_data: dict, database: Database = None):
     document_id = convert_document_url_to_id(document_url)
-
-    document_to_store = {
-        "document_id": document_id,
-        "document_url": document_url,
-        "document_type": document_type,
-        "extracted_data": extracted_data,
-        "status": "complete",
-    }
-
-    database.write_document(document_to_store)
+    document_item = DocumentItem(document_id, document_url, "complete", document_type, extracted_data)
+    database.write_document(document_item_to_dict(document_item))
```

Comment:
I think we should pass in the `DocumentItem` into, and out of, the database.  So it is up to the database implementation to call `document_item_to_dict`.